### PR TITLE
#170 nits and etc

### LINKED ISSUE #170: Nits and etc.
I think the draft would enormously benefit the implementer if you had more sequence/protocol diagrams showing the information that is exchanged as you introduced the different options. You have a few of them (mostly at the beginning and towards the end with handshaking and tearing down the connection, however, it would have been easier to visualise every and each procedure with a successful and (maybe) an error handling scenario).

### PR DESCRIPTION
**Pull request by shjohnson-bt**
First stab at adding MP_PRIO example to MP_CONFIRM section as well as adding a new section on "Address knowledge exchange" containing further examples.

Hopefully I have interpreted the descriptions of MP_CONFIRM.MP_PRIO/MP_ADDADDR and MP_REMOVEADDR correctly!

Closes #170

### CODE REVIEW

**Review by markusa — APPROVED** [review]

**Review by NathalieRM — COMMENTED** [review]
Hi @shjohnson-bt
 
The structure of the diagrams and the description of the cases look quite good to me; however, I have a problem with the use of DCCP-Request and DCCP-Response packets to illustrate the transmission and confirmation of the mp_options. Given that the draft doesn't specify over which packet types should or must be sent these options, in theory, it is possible that an MP_PRIO option is sent over a DCCP-Request packet, but the scenario where the MP_PRIO is updated twice is not possible with 2 DCCP-Request packets (the second one would attempt to establish a new subflow). Furthermore, the draft only specifies that Data and Data-Ack packets must have MP_SEQ numbers, so, a DDCP-Request might or might not have a MP-SEQ number (which makes impossible the confirmation). 

In summary, my proposal would be to specify somewhere in he draft that these options (MP_PRIO, MP_ADDADDR, MP_REMOVEADDR and MP_CONFIRM) must be transmitted over Data or Data-Ack packets and modify this section accordingly.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 1]
```diff
@@ -1199,6 +1245,68 @@ handshake is as follows:
...
+* an identifier (id 2) for the IP address to remove (A2_IP) and which was specified in a previous MP_ADDADDR message.
+
+This DCCP-Request must also supply:
+* the leftmost 20 bytes of the HMAC(A) generated during the initial handshaking procedure described in {{handshaking}} and {{MP_HMAC}}
+* the sequence number (seqno 33) for this message
+
+Host B acknowledges receipt of the MP_REMOVEADDR message with a DCCP-Response containing the MP_CONFIRM option. The parameters supplied in this
+response are as follows:
+* an MP_CONFIRM containing the sequence number (seqno 33) of the request that we are responding to together with the MP_REMOVEADDR option
+* the leftmost 20 bytes of the HMAC(B) generated during the initial handshaking procedure {{handshaking}}
```
Comment (NathalieRM) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1293]:
```suggestion
Host B acknowledges receipt of the MP_REMOVEADDR message with a DCCP-Ack containing the MP_CONFIRM option. The parameters supplied in this
response are as follows:
* an MP_CONFIRM containing the MP_SEQ number (seqno 33) of the packet carrying the option that we are confirming, together with the MP_REMOVEADDR option
* the leftmost 20 bytes of the HMAC(B) generated during the initial handshaking procedure {{handshaking}}
```

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 2]
```diff
@@ -1199,6 +1245,68 @@ handshake is as follows:
...
+          Host A                                         Host B 
+------------------------                              -----------
+Address A1    Address A2                               Address B1
+----------    ----------                              -----------
+     |             |                                       |
+     |   DCCP-Request +  MP_REMOVEADDR(id 2) +             |
+     |------- MP_HMAC(A) + MP_SEQ(seqno 33) -------------->|
+     |             |                                       |      
+     |   DCCP-Response + MP_HMAC(B) +                      |
+     |<----- MP_CONFIRM(seqno 33, MP_REMOVEADDR) ----------|
```
Comment (NathalieRM) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1305]:
```suggestion
     |             |                                       |
     |   DCCP-Data +  MP_REMOVEADDR(id 2) +             |
     |------- MP_HMAC(A) + MP_SEQ(seqno 33) -------------->|
     |             |                                       |      
     |   DCCP-Ack + MP_HMAC(B) +                      |
     |<----- MP_CONFIRM(seqno 33, MP_REMOVEADDR) ----------|
```

**Review by NathalieRM — CHANGES_REQUESTED** [review]
I have reconsidered the use of DCCP-Request and DCC-Response, and I think at least in the first example for the MP_RRIO option, it is ok to use it (the use of MP_SEQ is mandatory for MP_PRIO, so you can ignore my previous concern about it), for the other scenarios I have made suggestions to change them to DCCP-Data and DCCP-Ack packets

**Comment by shjohnson-bt** [comment]
Thanks Nathalie – the corrections make sense. I have accepted them and requested a re-review.

Kind Regards,
Stephen

**Comment by abrunstrom** [comment]
@NathalieRM Is this pull request ready? I think you requested changes but have not approved the updated version.

**Review by NathalieRM — APPROVED** [review]

**Comment by NathalieRM** [comment]
Hi @abrunstrom, 

The pull request is ready. I just approved it.

**Comment by abrunstrom** [comment]
Hi @markusa I saw you merged this pull request. I do not think Simone had reviewed it? My question to Nathalie was to understand if it was ready for review.

**Comment by markusa** [comment]
@abrunstrom Correct, she did not confirm it here, but in the corresponding issue #170. That's why I merged the PR. Do you think that was wrong?

**Comment by abrunstrom** [comment]
I understood the comment in #170 as "the plan looks good", but not sure she looked at the PR. She asked me if it was ready as Nathalie had requested changes and had not approved it yet. I have pinged her to find out, she can still look at the diff.

**Comment by abrunstrom** [comment]
Simone has now also had a look at the updates. "The sequence diagrams that were added are a huge improvement", so well done :)

So all good to go now @markusa.

**Comment by markusa** [comment]
Thank you @abrunstrom
